### PR TITLE
Adds AI audiotranscript correction and simple tools to interact with it

### PR DESCRIPTION
**Pull request by holoisme**
Hello,
This PR adds support for OpenAI's GPT-4o mini in our Firebase Cloud Functions, use the `prompt` function to interact with it.
A first clear example is in the `addAudioTranscript` cloud function where we prompt it to correct the audio transcript with some added context.

`prompt` takes in a task (the "order" with the highest level of authority), the content (additional content with user level authority) and a fallback value in case of an error.

### CODE REVIEW

File: edweiss-firebase/functions/src/functions/course/toogleCourse.ts [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
-import { CollectionOf, getDocumentAndRef } from 'utils/firestore';
-import { ok } from 'utils/status';
+import { fail } from 'utils/status';
 
 
 export const toogleCourse = onAuthentifiedCall(Course_functions.Functions.toogleCourse, async (userId, args) => {
-	if (!args.courseID && !userId) {
-		return ok({});//fail("empty_course_id");
-	}
+	return fail("don't use this.")
```
Comment (FerdinandGomerChiu) [inline edweiss-firebase/functions/src/functions/course/toogleCourse.ts:12]: Why should we use toggleCourse?

File: edweiss-app/model/lectures/lectureDoc.ts [hunk 2]
```diff
@@ -10,37 +10,36 @@
...
-        pdfUri: string;
-        nbOfPages: number;
-        availableToStudents: boolean;
-        audioTranscript: { [pageNumber: number]: string; };
-    }
-
-    export const Functions = FunctionFolder("lectures", {
-        addAudioTranscript: FunctionOf<{ courseId: string, lectureId: string, pageNumber: number, transcription: string; }, {}, 'invalid_arg' | 'error_firebase' | 'successfully_added'>("addAudioTranscript"),
-    });
+	export type LectureEvents = "quiz";
```
Comment (FerdinandGomerChiu) [inline edweiss-app/model/lectures/lectureDoc.ts:20]: What was the change here? Some code is highlighted in green, even though I do not see any changes with the last version (in red).

Comment (holoisme) [reply edweiss-app/model/lectures/lectureDoc.ts:20]: It's probably tabulations vs spaces.

**Review by FerdinandGomerChiu — APPROVED** [review]
Great work all around, some changes are confusing.

**Comment by holoisme** [comment]
> Great work all around, some changes are confusing.

Can you tell me what changes are confusing you ?

**Comment by FerdinandGomerChiu** [comment]
> > Great work all around, some changes are confusing.
> 
> Can you tell me what changes are confusing you ?

I changed my comment, I didn't see your replies at the time.

**Pull request merged**